util/src/main/java/tc/oc/pgm/util/tablist/PlayerTabEntry.java

```diff
@@ -78,4 +79,9 @@ protected void onSkinPartsChange(PlayerSkinPartsChangeEvent event) {
   public String toString() {
     return this.getClass().getSimpleName() + "{" + this.player.getName() + "}";
   }
+
+  @Override
+  public int getPing() {
+    return ((CraftPlayer) this.player).getHandle().ping;
+  }
```

Comment:
`this.player` is only updated on each `PlayerSkinPartsChangeEvent`. To keep it consistent you would have to request the ping form the active player object at a set interval.